dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/datasetreport/jdbc/AnalyticsDataSetReportStore.java

```diff
@@ -243,9 +243,33 @@ public AnalyticsDataSetReportStore( DataQueryService dataQueryService, Analytics
         for ( Entry<String, Object> entry : map.entrySet() )
         {
             String[] split = entry.getKey().split( SEPARATOR );
+            addToMap( dataMap, split[0], entry.getValue() );
             dataMap.put( split[0], entry.getValue() );
         }
 
         return dataMap;
     }
+
+    // -------------------------------------------------------------------------
+    // Supportive methods
+    // -------------------------------------------------------------------------
+
+    private void addToMap( Map<String, Object> dataMap, String key, Object value )
+    {
+        if ( value == null )
+        {
+            return;
+        }
+
+        Object existingValue = dataMap.get( key );
+
+        if ( existingValue == null || !(existingValue instanceof Double) || !(value instanceof Double) )
+        {
+            dataMap.put( key, value );
+        }
+        else
+        {
+            dataMap.put( key, (Double) existingValue + (Double) value );
+        }
```

Comment:
Could use `Map#compute` instead

Reply:
I hadn't used `Map#compute` before. Thanks for the tip. Done.